src/main/java/edu/iu/terracotta/service/app/impl/AssessmentServiceImpl.java

```diff
@@ -682,11 +682,18 @@ public AssessmentDto viewAssessment(long experimentId, SecuredInfo securedInfo)
 
     @Override
     public void verifySubmissionLimit(Integer limit, int existingSubmissionsCount) throws AssignmentAttemptException {
-        if (limit == null || limit == 0) {
+        if (limit == null && existingSubmissionsCount == 0) {
```

Comment:
So this is a change here in what a null limit means. Previously it meant unlimited. But I think that is fine to change it to meaning multiple attempts aren't allowed. Previously created Assessments have a null numOfSubmissions, so I think that's a good backward compatible interpretation of null.

One request though, please update the comment on https://github.com/terracotta-education/terracotta/blob/535442fc10ffa6632d6242344940f70d07426e8e/src/main/java/edu/iu/terracotta/model/app/Assessment.java#L48 and https://github.com/terracotta-education/terracotta/blob/2e283793c2fcc6c3931597c91bf910220c2942e4/src/main/java/edu/iu/terracotta/model/app/Assignment.java#L52 so it's documented that null means no multiple attempts and 0 means unlimited.